solidity/contracts/bridge/Wallets.sol

```diff
@@ -299,4 +304,35 @@ library Wallets {
 
         emit NewWalletRegistered(ecdsaWalletID, walletPubKeyHash);
     }
+
+    /// @notice Reports about a proven fraud committed by the given wallet.
+    /// @param walletPubKeyHash 20-byte public key hash of the wallet
+    /// @dev Requirements:
+    ///      - Wallet must be in Live state
+    function notifyWalletFraudProven(
+        Data storage self,
+        bytes20 walletPubKeyHash
+    ) external {
+        Wallet storage wallet = self.registeredWallets[walletPubKeyHash];
+        require(
+            wallet.state == WalletState.Live,
```

Comment:
What if the fraud is proven when the wallet is in the state `MovingFunds`?
We allow to submit a fraud challenge in this state.

Reply:
Good catch! See https://github.com/keep-network/tbtc-v2/pull/177/commits/21bafb8624066b2cd0113fa2fd1922685c1515a4